src/openrct2/core/MemoryStream.h

```diff
@@ -58,7 +58,51 @@ class MemoryStream final : public IStream
     void Seek(int64_t offset, int32_t origin) override;
 
     void Read(void* buffer, uint64_t length) override;
+    void Read1(void* buffer) override;
+    void Read2(void* buffer) override;
+    void Read4(void* buffer) override;
+    void Read8(void* buffer) override;
+    void Read16(void* buffer) override;
+
+    template<typename T> void Read(T& buffer)
+    {
+        uint64_t position = GetPosition();
+        if (position + sizeof(T) > _dataSize)
+        {
+            throw IOException("Attempted to read past end of stream.");
+        }
+
+        buffer = *reinterpret_cast<T*>(_position);
```

Comment:
This is illegal, as you end up with misaligned pointers.

Reply:
I'm not sure I see the issue, if you are at offset 1 and read 4 bytes via normal memcpy you have the same access pattern.